sklearn/linear_model/_base.py

```diff
@@ -387,7 +387,9 @@ def predict(self, X):
     def _set_intercept(self, X_offset, y_offset, X_scale):
         """Set the intercept_"""
         if self.fit_intercept:
-            self.coef_ = self.coef_ / X_scale
+            # We always want coef_.dtype=X.dtype. For instance, X.dtype can differ from
+            # coef_.dtype if warm_start=True.
+            self.coef_ = np.divide(self.coef_, X_scale, dtype=X_offset.dtype)
```

Comment:
Can we use `X_scale` here? (It reads slightly nicer since we are dividing by it)

```suggestion
            self.coef_ = np.divide(self.coef_, X_scale, dtype=X_scale.dtype)
```